fs2/config/__init__.py

```diff
@@ -281,4 +292,22 @@ def load_config_from_path(path: Path) -> "FastSpeech2Config":
             config = FastSpeech2Config(**config)
         return config
 
+    @model_validator(mode="before")
+    @classmethod
+    def check_and_upgrade_checkpoint(cls, data: Any) -> Any:
+        """
+        Check model's compatibility and possibly upgrade.
+        """
+        ckpt_version = data.get("VERSION", "0.0")
+        if ckpt_version > FastSpeech2Config_latest_version:
+            raise ValueError(
+                "Your config was created with a newer version of EveryVoice, please update your software."
+            )
+        # Successively convert model checkpoints to newer version.
+        if ckpt_version < "1.0":
```

Comment:
Why not just use `data.get("VERSION", "1.0")` ? Maybe I'm misunderstanding, but we're treating all previous model configs as 0.0 but once we add the VERSION info, they become 1.0?